src/classes/QueryPredicate.cls

```diff
@@ -0,0 +1,60 @@
+/*************************************************************************************************
+* This file is part of the Nebula Framework project, released under the MIT License.             *
+* See LICENSE file or go to https://github.com/jongpie/NebulaFramework for full license details. *
+*************************************************************************************************/
+public class QueryPredicate {
```

Comment:
we have switched to using get for pretty much everything because we are often constructing objects with several repos and passing their interfaces in accordingly, which obviates the need for having specific types in the objects being constructed themselves - but that would be pretty hard to manage if we had to keep on adding methods to the base interface. the query predicate encapsulates the request nicely, and the repo is just responsible for query / dml operations. ALSO (bonus): you can pass an empty query predicate to get everything (which we have a use case for: querying a very small object that serves as a joint object
between User and sales people that serves as the master in a master detail relationship)